Mon-32851 [gorgone] service auto-discovery only the services associated to one host are created (#3160)
centreon/centreon: centreon-gorgone/docs/modules/centreon/autodiscovery.md

    \"dry_run\": 1
}"
```

### Developer manual

> This module heavily use the gorgone-action module to work.

Reviewer: ```suggestion
This module heavily uses the gorgone-action module to work.

```

> here is a diagram of how these module interact : 

Reviewer: ```suggestion
Here is a diagram of how these modules interact: 

```

![image](./centreon-gorgone-autodiscovery-archi.png)

> dotted line mean a ZMQ message is send, direct line mean the function is called normally.

Reviewer: ```suggestion
Dotted lines mean a ZMQ message is sent. Direct lines mean the function is called normally.

```

> Each column represent a linux thread, as Gorgone is multi-threaded.

Reviewer: ```suggestion
Each column represents a Linux thread, as Gorgone is multi-threaded.

```

For each ZMQ message, names are described in the [events section](#events) of each module,
and for putlog the second part is the 'code' used by gorgone-autodiscovery 

> and defined as constant in the [class.pm](../../../gorgone/modules/centreon/autodiscovery/class.pm) file

Reviewer: ```suggestion
and defined as constant in the [class.pm](../../../gorgone/modules/centreon/autodiscovery/class.pm) file.

```

> gorgone-action module don't directly send the result to the calling module. Instead it send a putlog message, processed by core.\

Reviewer: ```suggestion
The gorgone-action module does not send the result directly to the calling module. It sends a putlog message instead, processed by core.\
```

> core keep track  of every module waiting for a particular event (use library.pm::addlistener to show interest in an event)\

Reviewer: ```suggestion
Core keeps track of every module waiting for a particular event (using library.pm::addlistener to show interest in an event)\

```
